### PR TITLE
[TT-9891] Improved clarity on key lifetime controls

### PR DESCRIPTION
**Pull request by andyo-tyk**
### Preview Link
https://deploy-preview-3355--tyk-docs.netlify.app/docs/nightly/basic-config-and-security/security/authentication-authorization/physical-key-expiry/

### Description
Improved explanation of the interactions between `session_lifetime`, `global_session_lifetime`, `force_global_session_lifetime` and `session_lifetime_respects_key_expiration`.

## Checklist

- [x ] I have added a **preview link** to the PR description.
- [x ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ x] Make sure you have started *your change* off *our latest `master`*.
- [ x] I **labelled** the PR
         -> [Something else (please add if needs merging to production or not)

### CODE REVIEW

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md [hunk 1]
```diff
@@ -57,10 +57,21 @@ This feature works nicely with [JWT]({{< ref "basic-config-and-security/security
 
 You can set a global lifetime for all keys created in the Redis by setting [global_session_lifetime]({{< ref "tyk-oss-gateway/configuration#global_session_lifetime" >}}) in the `tyk.conf` file; this parameter is an integer value in seconds.
 
-It is important to note that `session_lifetime` usually takes precedence over `global_session_lifetime`, so the gateway-level control will only apply if no per-API lifetime has been configured.
+To enable this global lifetime, you must also set the [force_global_session_lifetime]({{< ref "tyk-oss-gateway/configuration#force_global_session_lifetime" >}}) parameter in the `tyk.conf` file.
+
+### Summary of key lifetime (deletion) configuration
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md]:
```suggestion
### Summary of key lifetime precedence
```

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md [hunk 2]
```diff
@@ -57,10 +57,21 @@ This feature works nicely with [JWT]({{< ref "basic-config-and-security/security
...
+
+| `force_global_session_lifetime` | `session_lifetime_respects_key_expiration` | Lifetime assigned to key (in seconds)     |
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md]:
```suggestion
| `force_global_session_lifetime` | `session_lifetime_respects_key_expiration` | Assigned lifetime |
```

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md [hunk 3]
```diff
@@ -57,10 +57,21 @@ This feature works nicely with [JWT]({{< ref "basic-config-and-security/security
 
 You can set a global lifetime for all keys created in the Redis by setting [global_session_lifetime]({{< ref "tyk-oss-gateway/configuration#global_session_lifetime" >}}) in the `tyk.conf` file; this parameter is an integer value in seconds.
 
-It is important to note that `session_lifetime` usually takes precedence over `global_session_lifetime`, so the gateway-level control will only apply if no per-API lifetime has been configured.
+To enable this global lifetime, you must also set the [force_global_session_lifetime]({{< ref "tyk-oss-gateway/configuration#force_global_session_lifetime" >}}) parameter in the `tyk.conf` file.
+
+### Summary of key lifetime (deletion) configuration
+
```
Comment (dcs3spp) [inline tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md:63]:
```suggestion

The table below shows the key lifetime assigned for the different permutations of `force_global_session_lifetime` and  `session_lifetime_respects_key_expiration` configuration parameters.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR looks good to me. I like the idea of including the table for assigned key lifetime for the different configurations!

Added couple of minor suggestions and once a peer review has been approved I'll merge and release.

**Review by dcs3spp — APPROVED** [review]

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md [hunk 4]
```diff
@@ -57,10 +57,22 @@ This feature works nicely with [JWT]({{< ref "basic-config-and-security/security
...
+| `false`                         | `true`                                     | larger of `session_lifetime` or `expires` |
+| `false`                         | `false`                                    | `session_lifetime`                        |
 
 {{< note success >}} 
 **Note**
 
-You can override the per-API lifetime with the global lifetime setting using the [force_global_session_lifetime]({{< ref "tyk-oss-gateway/configuration#force_global_session_lifetime" >}}) parameter in the `tyk.conf` file.
+It is important to remember that a value of `0` in `session_lifetime` or `global_session_lifetime` is interpreted as infinity (i.e. key will not be deleted if that control is in use) - and if a field is not set, this is treated as `0`.
+<br>
+If you want to the key to be deleted when it expires (i.e. to use the expiry configured in `expires` within the key to control deletion) then you must set a non-zero value in `session_lifetime` and configure both `session_lifetime_respects_key_expiration:true` and `force_global_session_lifetime:false`.
```
Comment (furkansenharputlu) [inline tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md]: shouldn't this be "if you want the key to be deleted..." ?

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md [hunk 5]
```diff
@@ -57,10 +57,22 @@ This feature works nicely with [JWT]({{< ref "basic-config-and-security/security
 
 You can set a global lifetime for all keys created in the Redis by setting [global_session_lifetime]({{< ref "tyk-oss-gateway/configuration#global_session_lifetime" >}}) in the `tyk.conf` file; this parameter is an integer value in seconds.
 
-It is important to note that `session_lifetime` usually takes precedence over `global_session_lifetime`, so the gateway-level control will only apply if no per-API lifetime has been configured.
+To enable this global lifetime, you must also set the [force_global_session_lifetime]({{< ref "tyk-oss-gateway/configuration#force_global_session_lifetime" >}}) parameter in the `tyk.conf` file.
```
Comment (furkansenharputlu) [inline tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md:60]: I don't know why but when I click the `Display rich diff` button to preview the changes the links are not working. Is it a problem in the display mode or actually a misuse of link notation?

**Review by furkansenharputlu — COMMENTED** [review]
Added minor comments. Technical-wise, LGTM.

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md (hunk 4 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md]: Oops 🤦

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md (hunk 5 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md:60]: I think it's a Hugo thing - you need to go via the Netlify [link](https://deploy-preview-3355--tyk-docs.netlify.app/docs/nightly/basic-config-and-security/security/authentication-authorization/physical-key-expiry/) to check links.

File: tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md (hunk 4 above)
Comment (andyo-tyk) [reply tyk-docs/content/basic-config-and-security/security/authentication-authorization/physical-key-expiry.md]:
```suggestion
If you want the key to be deleted when it expires (i.e. to use the expiry configured in `expires` within the key to control deletion) then you must set a non-zero value in `session_lifetime` and configure both `session_lifetime_respects_key_expiration:true` and `force_global_session_lifetime:false`.
```

**Review by furkansenharputlu — APPROVED** [review]

**Pull request merged**